youtube_dl/extractor/vidlii.py

```diff
@@ -46,9 +67,31 @@ def _real_extract(self, url):
         webpage = self._download_webpage(
             'https://www.vidlii.com/watch?v=%s' % video_id, video_id)
 
-        video_url = self._search_regex(
-            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1', webpage,
-            'video url', group='url')
+        formats = []
+
+        def add_format(format_url, height=None):
+            height = int(self._search_regex(r"(\d+).mp4",
+                         format_url, "height", default=360))
+
+            formats.append({
+                'url': format_url,
+                'format_id': '%dp' % height if height else None,
+                'height': height,
+            })
+
+        sources = re.findall(
+            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
```

Comment:
If the attribute names are all `src` and not `xxxsrc`:
```suggestion
            r'\bsrc\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
```

Reply:
The attribute names are `hdsrc` and `src`.